src/tests/audio/data_structures/audio_test.py

```diff
@@ -17,19 +17,124 @@
 except ModuleNotFoundError:
     TORCHAUDIO_AVAILABLE = False
 
+try:
+    import soundfile
+
+    SOUNDFILE_AVAILABLE = True
+except ModuleNotFoundError:
+    SOUNDFILE_AVAILABLE = False
+
 
 def load_audio(file_path: str) -> Tuple[torch.Tensor, int]:
     """Loads audio data from the given file path."""
     return torchaudio.load(file_path)
 
 
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_full_file() -> None:
+    """Tests loading the full audio file without offset or duration."""
+    audio = Audio.from_filepath(MONO_AUDIO_PATH)
+    assert audio is not None
+    assert audio.waveform is not None
+    assert audio.waveform.shape[1] > 0
+    assert isinstance(audio.sampling_rate, int)
+    assert audio.sampling_rate == 48000
+
+
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_with_offset() -> None:
+    """Tests loading audio with a positive offset."""
+    audio = Audio.from_filepath(MONO_AUDIO_PATH, offset_in_sec=1.0)
+    assert audio is not None
+    assert audio.waveform is not None
+    assert audio.waveform.shape[1] > 0
+    assert isinstance(audio.sampling_rate, int)
+    assert audio.sampling_rate == 48000
+
+
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_with_offset_exceeding_duration() -> None:
+    """Tests that an excessive offset raises an error."""
+    info = torchaudio.info(MONO_AUDIO_PATH)
+    file_duration = info.num_frames / info.sample_rate
+
+    with pytest.raises(ValueError, match="Offset .* exceeds the duration of the audio file"):
+        Audio.from_filepath(MONO_AUDIO_PATH, offset_in_sec=file_duration + 1.0)
+
+
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_with_duration() -> None:
+    """Tests loading a specific duration of an audio file."""
+    audio = Audio.from_filepath(MONO_AUDIO_PATH, duration_in_sec=2.0)
+    assert audio is not None
+    assert audio.waveform is not None
+    assert audio.waveform.shape[1] > 0
```

Comment:
this one is actually very easy to check that the duration is equal to 2 sampling rates